src/rapi/rc_api_info.c

```diff
@@ -492,3 +492,91 @@ int rc_api_process_fetch_game_titles_server_response(rc_api_fetch_game_titles_re
 void rc_api_destroy_fetch_game_titles_response(rc_api_fetch_game_titles_response_t* response) {
   rc_buffer_destroy(&response->response.buffer);
 }
+
+/* --- Fetch Game Hashes --- */
+
+int rc_api_init_fetch_hash_library_request(rc_api_request_t* request,
+                                           const rc_api_fetch_hash_library_request_t* api_params)
+{
+  return rc_api_init_fetch_hash_library_request_hosted(request, api_params, &g_host);
+}
+
+int rc_api_init_fetch_hash_library_request_hosted(rc_api_request_t* request,
+                                                  const rc_api_fetch_hash_library_request_t* api_params,
+                                                  const rc_api_host_t* host)
+{
+  rc_api_url_builder_t builder;
+  rc_api_url_build_dorequest_url(request, host);
+
+  /* note: unauthenticated request */
+  rc_url_builder_init(&builder, &request->buffer, 48);
+  rc_url_builder_append_str_param(&builder, "r", "hashlibrary");
+  rc_url_builder_append_unum_param(&builder, "c", api_params->console_id);
```

Comment:
It's more correct to not send the `c` parameter instead of sending 0:
```
if (api_params->console_id != RC_CONSOLE_UNKNOWN)
  rc_url_builder_append_unum_param(&builder, "c", api_params->console_id);
```